### PR TITLE
Show different message when you can't skip an ad.

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

Apologies for the double whammy.
- Show prerolls on video pages again as we were just resolving the promise with `shouldHideAdverts: true`, whereas we should have been looking it up.
- Only add the skip timer when the ad is longer than the time to skip.
## What is the value of this and can you measure success?
- We get no more complaints saying how can I skip if the ad ends before I can skip.
- We serve preroll again.
## Does this affect other platforms - Amp, Apps, etc?

No
## Request for comment

@akash1810

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 1]
```diff
@@ -185,6 +185,7 @@ define([
             embedPath = $el.attr('data-embed-path'),
             // we need to look up the embedPath for main media videos
             canonicalUrl = $el.attr('data-canonical-url') || (embedPath ? '/' + embedPath : null),
+            shouldHideAdverts = $el.attr('data-block-video-ads') === 'false' ? false : true,
```
Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: this value isn't always correct, as embeds don't follow parent updates 😢 I guess its a good default value though...

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 2]
```diff
@@ -295,13 +296,19 @@ define([
...
                                     function () {
-                                        player.adSkipCountdown(15);
+                                        var skipAdTime = 15;
                                         player.ima({
                                             id: mediaId,
                                             adTagUrl: getAdUrl(),
                                             prerollTimeout: 1000
                                         });
                                         player.ima.requestAds();
+                                        player.on('adstart', function() {
```
Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: should the event handler be before requesting the ad?

**Comment by akash1810** [comment]
when ads are < 15 seconds long, should there be a count down? Such as "the video will play in 5 seconds"?

File: static/src/javascripts/bootstraps/enhanced/media/main.js (hunk 1 above)
Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: Yea, so, look it up, if you can't find it, set to true.
With embeds we look it up though.

File: static/src/javascripts/bootstraps/enhanced/media/main.js (hunk 2 above)
Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: I hate that that matters, but you're probably right. Changed.

**Comment by jamesgorrie** [comment]
> when ads are < 15 seconds long, should there be a count down? Such as "the video will play in 5 seconds"?

There's a progress bar.

File: static/src/javascripts/bootstraps/enhanced/media/main.js (hunk 1 above)
Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: Order:

* has attribute? read attribute => set `shouldHideAdverts`
* has canonical URL? look it up via `info.json` => set `shouldHideAdverts`

So we should be right nearly all of the time except, like the other day, the `info.json` borks.

**Comment by akash1810** [comment]
👍

**Comment by jamesgorrie** [comment]
> when ads are < 15 seconds long, should there be a count down? Such as "the video will play in 5 seconds"?

@akash1810 maybe that is nicer. Updated:
timer

File: static/src/javascripts/projects/common/modules/video/events.js [hunk 3]
```diff
@@ -233,10 +233,15 @@ define([
...
+                    var adDuration = this.ima.getAdsManager().getCurrentAd().getDuration();
+
+                    var skipButton = template(adsSkipOverlayTmpl, {
+                        adDuration: adDuration,
+                        skipTimeout: skipTimeout
+                    });
 
                     $(this.el()).append(skipButton);
-                    intervalId = setInterval(events.update.bind(this), 250);
+                    intervalId = setInterval(events.update.bind(this), 500);
```
Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/video/events.js]: This makes it seem more synced.

**Comment by akash1810** [comment]
Nice! 👍

**Pull request merged**